System/OsPath/Common.hs

```diff
@@ -1467,3 +1504,137 @@ decodeFS (OsString (PosixString x)) = decodeWithBasePosix x
 
 #endif
 
+#ifdef WINDOWS
```

Comment:
Looks like sometimes it's checked as `#ifdef WINDOWS` and other times the check looks like `#if defined(WINDOWS)`. I suggest to unify everything to `#if defined(WINDOWS)` style.